syntax = "proto3";

package pref;

service Preferans {
  rpc SayHello (HelloRequest) returns (HelloResponse);
  rpc Join (JoinRequest) returns (JoinResponse);
}

message Player {
  string player_id = 1;
  string player_name = 2;
}

message JoinRequest {
  string player_name = 1;
}

message JoinResponse {
  string player_id = 1;             // ID of the player who just joined
  repeated Player players = 2;      // Full list of connected players
}

message HelloRequest {
  string name = 1;
}

message HelloResponse {
  string message = 1;
}

message PlayerLeft {
  string player_id = 1;
}

message RpcEnvelope {
  string method = 1;
  bytes payload = 2;
}
